dipy/viz/actor.py

```diff
@@ -911,4 +1034,204 @@ def figure(pic, interpolation='nearest'):
         image_actor.GetProperty().SetInterpolationTypeToCubic()
 
     image_actor.Update()
-    return image_actor
\ No newline at end of file
+    return image_actor
+
+
+class Container(object):
+    """ Provides functionalities for grouping multiple actors using a given
+    layout.
+
+    Attributes
+    ----------
+    anchor : 3-tuple of float
+        Anchor of this container used when laying out items in a container.
+        The anchor point is relative to the center of the container.
+        Default: (0, 0, 0).
+
+    padding : 6-tuple of float
+        Padding around this container bounding box. The 6-tuple represents
+        (pad_x_neg, pad_x_pos, pad_y_neg, pad_y_pos, pad_z_neg, pad_z_pos).
+        Default: (0, 0, 0, 0, 0, 0)
+
+    """
+    def __init__(self, layout=layout.Layout()):
+        """
+        Parameters
+        ----------
+        layout : ``dipy.viz.layout.Layout`` object
+            Items of this container will be arranged according to `layout`.
+        """
+        self.layout = layout
+        self._items = []
+        self._need_update = True
+        self._position = np.zeros(3)
+        self._visibility = True
+        self.anchor = np.zeros(3)
+        self.padding = np.zeros(6)
+
+    @property
+    def items(self):
+        if self._need_update:
+            self.update()
+
+        return self._items
+
+    def add(self, *items, **kwargs):
+        """ Adds some items to this container.
+
+        Parameters
+        ----------
+        items : `vtkProp3D` objects
+            Items to add to this container.
+        borrow : bool
+            If True the items are added as-is, otherwise a shallow copy is
+            made first. If you intend to reuse the items elsewhere you
+            should set `borrow=False`. Default: True.
+        """
+        self._need_update = True
+
+        for item in items:
+            if not kwargs.get('borrow', True):
+                item = shallow_copy(item)
+
+            self._items.append(item)
+
+    def clear(self):
+        """ Clears all items of this container. """
+        self._need_update = True
+        del self._items[:]
+
+    def update(self):
+        """ Updates the position of the items of this container. """
+        self.layout.apply(self._items)
+        self._need_update = False
+
+    def add_to_renderer(self, ren):
+        """ Adds the items of this container to a given renderer. """
+        for item in self.items:
+            if isinstance(item, Container):
+                item.add_to_renderer(ren)
+            else:
+                ren.add(item)
+
+    def GetBounds(self):
+        """ Get the bounds of the container. """
+        bounds = np.zeros(6)    # x1, x2, y1, y2, z1, z2
+        bounds[::2] = np.inf    # x1, y1, z1
+        bounds[1::2] = -np.inf  # x2, y2, z2
+
+        for item in self.items:
+            item_bounds = item.GetBounds()
```

Comment:
`item` can be either a `vtkActor` or a `Container`. This is why `GetBounds` must be written this way. If we change the name we'll need to put a `if` statement everywhere we use those functions! Do you prefer the latter?